src/compiler/sys.ts

```diff
@@ -1186,7 +1186,8 @@ namespace ts {
             let activeSession: import("inspector").Session | "stopping" | undefined;
             let profilePath = "./profile.cpuprofile";
 
-            const realpathSync = _fs.realpathSync.native ?? _fs.realpathSync;
+            const useCaseSensitiveFileNames = isFileSystemCaseSensitive();

```

Comment:
I think you want to move `realpathSync` down instead - `isFileSystemCaseSensitive` depends on the initialization of `platform`.

Reply:
Oof, good catch!